### ISSUE TITLE
feature request: Reference credentials from a profile

### ISSUE THREAD

**Issue by cristim** [aws/aws-cli#1086]
In order to avoid duplication of credentials profiles, it would be nice to have a way to define credentials only once in the credentials file and then define multiple profiles in the config where the same set of credentials could be reused, and only augmented like in the example below, for example with the region information:

In .aws/credentials:

```
[example]
aws_access_key_id=foo
aws_secret_access_key=bar
```

in .aws/config

```
[profile example-us-east]                                                     
region=us-east-1
credentials=example

[profile example-eu-west]                                                     
region=eu-west-1
credentials=example

```

**Comment by garnaat** [comment]
Maybe extend the ``source_profile`` attribute used for IAM Role-based profiles?

**Comment by jamesls** [comment]
Could you share a little more about your use case?  Would just like to get more background on how this would be used.  The two _main_ config values besides the credentials are region and output.  Is the idea that you have one set of credentials but you want a different region per profile as shown above?

**Comment by cristim** [comment]
@jamesls exactly.

Let's say I have an AWS account with resources spread across 4(or more) regions. The idea is to have a single credentials profile containing my API keys, and have it referenced by 4 different configuration profiles, every one just defining its region and pointing to the shared credentials.

The benefit would be slightly easier management of access keys.

**Comment by jamesls** [comment]
Was there an issue you were running into just using the `--region` argument?  It seems like the difference would be:

```
for region in $(aws ec2 describe-regions --query Regions[].RegionName --output text);
    do aws ec2 describe-instances --region $region
done
```

vs. 

```
for region in $(aws ec2 describe-regions --query Regions[].RegionName --output text);
    do aws ec2 describe-instances --profile "example-$region"
done
```

Marking as a feature request, but before adding I want to ensure that this solves a problem that is either currently not possible or very difficult to accomplish with the current feature set.

**Comment by jamesls** [comment]
Removing closing soon label.  I like the idea of generalizing `source_profile`.

**Comment by ASayre** [comment]
Good Morning!

We're closing this issue here on GitHub, as part of our migration to [UserVoice](https://aws.uservoice.com/forums/598381-aws-command-line-interface) for feature requests involving the AWS CLI.

This will let us get the most important features to you, by making it easier to search for and show support for the features you care the most about, without diluting the conversation with bug reports.

As a quick UserVoice primer (if not already familiar): after an idea is posted, people can vote on the ideas, and the product team will be responding directly to the most popular suggestions.

We’ve imported existing feature requests from GitHub - Search for this issue there!

And don't worry, this issue will still exist on GitHub for posterity's sake.  As it’s a text-only import of the original post into UserVoice, we’ll still be keeping in mind the comments and discussion that already exist here on the GitHub issue.

GitHub will remain the channel for reporting bugs.  

Once again, this issue can now be found by searching for the title on: https://aws.uservoice.com/forums/598381-aws-command-line-interface 

-The AWS SDKs & Tools Team

**Comment by jamesls** [comment]
Based on community feedback, we have decided to return feature requests to GitHub issues.

**Comment by pauldraper** [comment]
Does this issue indeed include the source_profile chaining mentioned in #1794

**Comment by darrikonn** [comment]
+1 on this. A simple implementation can be found here: https://github.com/boto/botocore/pull/1816

Basically introducing a new config variable called `source_credentials`, since `source_profile` is bound to a role being defined.
